File: Diagrammatize.Design.Pattern/code/cxx/Singleton/A2/Triple.cpp
#include "Triple.hpp"
#include <iostream>

Triple Triple::triple[3]{Triple(0), Triple(1), Triple(2)};

Triple::Triple(int id_): id(id_) {
    std::cout << "The instance " << id << " is created." << std::endl;
}

Triple& Triple::getInstance(int id) {
    return triple[id];
}

std::string Triple::toString() const {
    return "[Triple id=" + std::to_string(id) + "]";
}

std::ostream& operator<<(std::ostream& os, const Triple& triple) {
    os << triple.toString();
    return os;
}

File: Diagrammatize.Design.Pattern/code/python/Singleton/A2/Triple.py

class Triple:
    _triple = None

    def __init__(self, id_):
        print("The instance " + str(id_) + " is created.");
        self.id = id_ 

    def __str__(self):
        return "[Triple id=" + str(self.id) + "]";

    @classmethod
    def getInstance(cls, id_):
        if not cls._triple:
            cls._triple = [Triple(0), Triple(1), Triple(2)]
        return cls._triple[id_]
